lib/browser/guest-window-manager.js

```diff
@@ -78,19 +80,8 @@ const setupGuest = function (embedder, frameName, guest, options) {
     embedder.send('ELECTRON_GUEST_WINDOW_MANAGER_WINDOW_CLOSED_' + guestId)
     embedder.removeListener('render-view-deleted', closedByEmbedder)
   }
-  if (!options.webPreferences.sandbox) {
-    // These events should only be handled when the guest window is opened by a
-    // non-sandboxed renderer for two reasons:
```

Comment:
> Is this check only needed for opening sandboxed windows opened from non-sandbox windows?

I think it was related to opening sandboxed windows, regardless of the parent. Could also apply to `nativeWindowOpen`.

> On master, if you open a sandbox window then do window.open() the listeners did appear to be registered since the options have no web preferences by default so options.webPreferences.sandbox is false.

Since this code was first added, there were some changes to allow sandbox reuse code from non-sandbox, so it is possible that this is obsolete.

If I remember correctly, you could reproduce the crash by closing a window created by `window.open` and waiting a bit for the garbage collector. If you can't reproduce this now, I guess it should be fine to remove it.